pkg/cluster_agent/epsagon_client.py

```diff
@@ -0,0 +1,65 @@
+"""
+Async Epsagon client
+"""
+from http import HTTPStatus
+from aiohttp.client_exceptions import ClientError
+from aiohttp_retry import RetryClient, ExponentialRetry
+
+class EpsagonClientException(Exception):
+    pass
+
+
+class EpsagonClient:
+    """
+    Async Epsagon client
+    """
+
+    DEFAULT_RETRY_ATTEMPTS = 3
+
+    async def _initialize_client(self):
+        """
+        Initializes the client session.
+        """
+        self.client = RetryClient(
+            headers={
+                "Authorization": f"Basic {self.epsagon_token}",
+                "Content-Type": "application/json",
+            },
+            retry_options=self.retry_options,
+            raise_for_status=True
+        )
+
+    def __init__(
+            self,
+            epsagon_token,
+            retry_attempts=DEFAULT_RETRY_ATTEMPTS
+    ):
+        """
+        :param epsagon_token: used for authorization
+        """
+        self.epsagon_token = epsagon_token
+        self.retry_options = ExponentialRetry(
+            attempts=retry_attempts,
+            exceptions=(ClientError,)
+        )
+        self.client = None
+
+    async def post(self, url, data):
+        """
+        Posts data to Epsagon given url.
+        :param url: endpoint to post the data to
+        :param data: to send
+        HTTP status code.
+        """
+        if not self.client:
```

Comment:
why not just initialize_client in init, instead of checking each time post is called?

Reply:
I indeed tried that, but the client must be initialized under async function. I found another trick for that (see the classmethod create func - and https://stackoverflow.com/questions/33128325/how-to-set-class-attribute-with-await-in-init)